File: src/main/java/me/bristermitten/plumber/aspect/AspectAnnotation.java
package me.bristermitten.plumber.aspect;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks that an annotation is used by an aspect, and that the class scanner should initialize that Aspect if not
 * already done.
 */
@Target(ElementType.ANNOTATION_TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface AspectAnnotation {
    Class<? extends Aspect> target();
}

File: src/main/java/me/bristermitten/plumber/aspectold/AspectAnnotation.kt
package me.bristermitten.plumber.aspect

import me.bristermitten.plumber.newaspect.Aspect
import kotlin.reflect.KClass

/**
 * Marks that an annotation is used by an aspect, and that the class scanner should initialize that Aspect if not
 * already done.
 */
@Target(AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class AspectAnnotation(val target: KClass<out Aspect>)
